### DISCUSSION TITLE
Does Remix support Intercepted routes?

### DISCUSSION THREAD

**Discussion by lodybo** [remix-run/remix#8731] [General]
I was wondering if it's possible to do something like Next.js's [intercepted routes](https://nextjs.org/docs/app/building-your-application/routing/intercepting-routes) with Remix. 

I'm looking specifically for something like Facebook's way of handling images: when clicking an image it opens in a modal dialog, but the URL does change. If you browse the updated URL directly, you'd get a different page specifically designed for the image.

I was thinking of implementing this through middleware, but this seems a too specific use case for something like that. I could use some pointers in the right direction for this.

**Comment by kiliman** [comment]
I've seen an example but don't remember where at the moment. 

Basically the trick was to use `<Link state>` on your images then in the image route call `useLocation` on the detail page to see if the state is set and show as a modal.

**Reply by lodybo** [reply]
Ah, so the detail page would basically simulate the "previous page with a modal on top of it". That's pretty clever. Maybe a bit difficult with an infinitely scrolled page but with enough background blur and a skeleton that would be doable. I'm going to play around with this idea and report back.

**Reply by jrestall** [reply]
Maybe this https://github.com/jjenzz/insta-remix/

**Reply by kiliman** [reply]
@jrestall Thanks. That was the example I was thinking of.

**Reply by spacepolice10** [reply]
@lodybo any progress on this one? I found out that it is almost impossible to show element without significant changes in scrolling position. But maybe you solved this somehow? )
